pkg/scheduler/algorithm/generic.go

```diff
@@ -82,11 +82,7 @@ func (g *genericScheduler) Schedule(ctx context.Context, fwk framework.Framework
 	trace.Step("Computing predicates done")
 
 	if len(feasibleClusters) == 0 {
-		return result, &framework.FitError{
-			Subscription:   sub,
-			NumAllClusters: g.cache.NumClusters(),
-			Diagnosis:      diagnosis,
-		}
+		return result, nil
```

Comment:
Why changing this? It is irrelevant.

Reply:
For one boundary case, remove custom labels from all ManagedCluster objects. If nil is not returned here, then sub.status.bindingclusters will not be set to empty.

Reply:
The error here should not be set to `nil`. The error will be recorded as an event for `Subscription`.

> sub.status.bindingclusters will not be set to empty.

Then why not setting to nil in upper function?